Source/Dafny/Resolver/BitvectorOptimization.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright by the contributors to the Dafny Project
+// SPDX-License-Identifier: MIT
+
+using System;
+using System.Linq.Expressions;
+using System.Numerics;
+using Microsoft.Boogie;
+
+namespace Microsoft.Dafny;
+
+public class BitvectorOptimization : IRewriter {
+  public BitvectorOptimization(ErrorReporter reporter) : base(reporter) { }
+
+  internal override void PostResolveIntermediate(ModuleDefinition m) {
+    var visitor = new BitvectorOptimizationVisitor();
+    foreach (var decl in ModuleDefinition.AllCallables(m.TopLevelDecls)) {
+      visitor.Visit(decl);
```

Comment:
Will this miss cover everything (including consts?)

Reply:
I'll take a deeper look to make sure.

Reply:
Hmm. Constants are `ICallable` but `AllCallables` has specific logic to skip them? That's weird. Using `AllItersAndCallables` seems better. It doesn't skip constants and also includes `IteratorDecl` entries.